pkg/pipelines/utils.go

```diff
@@ -74,6 +77,12 @@ func BootstrapRepository(o *BootstrapOptions, f clientFactory, e executor) error
 }
 
 func pushRepository(o *BootstrapOptions, remote string, e executor) error {
+	exists, _ := ioutils.IsExisting(ioutils.NewFilesystem(), filepath.Join(o.OutputPath, ".git"))
```

Comment:
1.  We should only do so if --overwrite is true.
2.  If --overwrite is false and --push-to-git is true and .git exists, we should return a error message.   If possible, an error is return before we start performing "push to git" (i.e. we should not true to create a new git repo if --overwrite is false and .git exists)

Reply:
@ishitasequeira, are you assuming that if `--overwrite=false` and `output` directory exists then it would error out even before it tries to update the contents of the `output` directory? 
If that is true, i think its OK to not have the `--overwrite` check here!

Reply:
@wtam2018 As discussed today, I have addressed the comments. I have updated the PR description as well.
@shubhamagarwal19 I have updated the workflow to check for overwrite flag and `.git` directory and return an error before creating the github repository if the directory exists and overwrite flag is false.